### PR TITLE
US 1.8, 2.6 , 4.1, 4.3.1 fixes

### PR DESCRIPTION
**Pull request by haris-raheem**
## Description
US 1.8 
Fixed - When linking a Project Template via a Project Edit View the tasks are incorrectly calculated - see screenshot 1-8, 1-8-1, 18-2 of my steps. When creating a Project task from the project its correctly calculated. The Project Template itself is also selected to have working days considered i.e. true.

US 2.6 
Fixed - Unable to Create Project via Project Template
I can not create a Project via the Project Template > Edit button > Create Project. There is no error logs and I have already RepairJS files.

It worked from detail view but gantt chart view had problems which are fixed.

US 4.1, US 4.3.1
Fixed - it appears that the task only shows its details on the start and the end date blocks, but nothing between including the 'non' working days that I have selected (in the attachment case they were the Saturday and Sunday).

Fixed - Task Name is saving as 'undefined'.  
issue occurred because task name field got renamed by other users commits.

AC 4.1, AC4.3.1
Fixed - now the tooltips show only the selected projects.

US 2.6
Fix for Project Template > Edit Button > Create Project > Enter Popup information and Save.
PHP Fatal error: Call to a member function format() on a non-object

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [ ] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [ ] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

File: modules/AM_ProjectTemplates/controller.php [hunk 1]
```diff
@@ -1,436 +1,514 @@
...
+
+				if($bhours[$day] != 0 ){
+					$d += 1;	
+				}
+				$enddate = $enddate->modify('+1 Days');
+			} 
+			$enddate = $enddate->modify('-1 Days');//readjust it back to remove 1 additional day added
+
+			}
+			catch(Exception $e){ die(var_dump($e)); };
```
Comment (samus-aran) [inline modules/AM_ProjectTemplates/controller.php]: Remove Debugging.

**Comment by samus-aran** [comment]
Code Review Done, will test.

**Comment by haris-raheem** [comment]
debugging removed.

**Comment by haris-raheem** [comment]
Fixed - Task name undefined issue when saving from gantt view.

**Comment by haris-raheem** [comment]
Fixed AC 4.1 and 4.3.1

**Comment by haris-raheem** [comment]
US 2.6 Fatal issue fixed, now you can test.

**Comment by haris-raheem** [comment]
Fixed additional days issue while creating Project from template
Fixed consider business hours issue while switching template from project edit view.
Fixed consider business hours issue while adding/updating task from project gantt view.

**Comment by haris-raheem** [comment]
AC 4.3.1 - Fixed resource calendar tooltips missing tasks issue.

**Comment by haris-raheem** [comment]
Fixed US 2.7,  US1.2

**Pull request merged**